src/Utils/StrictPlus.php

```diff
@@ -0,0 +1,233 @@
+
+		if ($name === '') {
+			throw new MemberAccessException("Cannot read a class '$class' property without name.");
+
+		} elseif (isset($methods[$m = 'get' . $uname]) || isset($methods[$m = 'is' . $uname])) { // property getter
+			if ($methods[$m] === 0) {
+				$methods[$m] = (new \ReflectionMethod($class, $m))->returnsReference();
+			}
+			if ($methods[$m] === TRUE) {
+				return $this->$m();
+			} else {
+				$val = $this->$m();
+				return $val;
+			}
+
+		} elseif (isset($methods[$name])) { // public method as closure getter
+			trigger_error("Accessing methods as properties via \$obj->$name is deprecated" . ObjectMixin::getSource(), E_USER_DEPRECATED);
+			$val = Callback::closure($this, $name);
+			return $val;
+
+		} elseif (isset($methods['set' . $uname])) { // strict class
+			throw new MemberAccessException("Cannot read a write-only property $class::\$$name.");
+
+		} else { // strict class
+			$rc = new \ReflectionClass($class);
+			$hint = ObjectMixin::getSuggestion(array_merge(
+				array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC)),
+				ObjectMixin::parseFullDoc($class, '~^[ \t*]*@property(?:-read)?[ \t]+(?:\S+[ \t]+)??\$(\w+)~m')
+			), $name);
+			throw new MemberAccessException("Cannot read an undeclared property $class::\$$name" . ($hint ? ", did you mean \$$hint?" : '.'));
+		}
+	}
+
+
+	/**
+	 * @return void
+	 * @throws MemberAccessException if the property is not defined or is read-only
+	 */
+	public function __set($name, $value)
+	{
+		$class = get_class($this);
+		$uname = ucfirst($name);
+		$methods = & ObjectMixin::getMethods($class);
+
+		if ($name === '') {
+			throw new MemberAccessException("Cannot write to a class '$class' property without name.");
+
+		} elseif (ObjectMixin::hasProperty($class, $name)) { // unsetted property
+			$this->$name = $value;
+
+		} elseif (isset($methods[$m = 'set' . $uname])) { // property setter
+			$this->$m($value);
+
+		} elseif (isset($methods['get' . $uname]) || isset($methods['is' . $uname])) { // strict class
+			throw new MemberAccessException("Cannot write to a read-only property $class::\$$name.");
+
+		} else { // strict class
+			$rc = new \ReflectionClass($class);
+			$hint = ObjectMixin::getSuggestion(array_merge(
+				array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC)),
+				ObjectMixin::parseFullDoc($class, '~^[ \t*]*@property(?:-write)?[ \t]+(?:\S+[ \t]+)??\$(\w+)~m')
+			), $name);
+			throw new MemberAccessException("Cannot write to an undeclared property $class::\$$name" . ($hint ? ", did you mean \$$hint?" : '.'));
+		}
+	}
+
+
+	/**
+	 * @return void
+	 * @throws MemberAccessException
+	 */
+	public function __unset($name)
+	{
+		$class = get_class($this);
+		if (!ObjectMixin::hasProperty($class, $name)) { // strict class
+			throw new MemberAccessException("Cannot unset the property $class::\$$name.");
+		}
```

Comment:
This seems to pretend that property was sucessfully unset, but it was not, getter on such property would still work, wouldn't it?